### PR TITLE
Custom icon added for Chef Server in left side nav

### PR DESCRIPTION
**Pull request by sachin-msys**
Signed-off-by: sachin-msys <sbacchav@msystechnologies.com>

### :nut_and_bolt: Description: What code changed, and why?
Custom icon added for chef server in left side nav.

### :chains: Related Resources
#3660 
### :+1: Definition of Done
Custom icon added for Chef Servers in left side nav
### :athletic_shoe: How to Build and Test the Change
1. `git fetch origin Sachin/chef-servers-left-nav-custom-icon`
2. `build components/automate-ui` 
3. Go to `Infrastructure` tab >>  `Chef Servers` its under `Chef-Server` feature flag.
### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
chef-custome-icon-before
chef-custome-icon-after

### CODE REVIEW

File: components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.html [hunk 1]
```diff
@@ -12,6 +12,7 @@
 <ng-template #sidebarEntryTemplate>
   <chef-icon class="sidebar-icon" *ngIf="icon"
     [style.transform]="'rotate(' + (iconRotation ? iconRotation : 0) + 'deg)'">{{icon}}</chef-icon>
+  <div *ngIf="customIcon === 'chef-servers'" class="chef-servers-icon"></div>
```
Comment (msorens) [inline components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.html:15]: Would prefer if this class did not need to know about its callers (that goes against the Open/Closed Principle -- the "O" in [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design))). Perhaps we could just do something like this:
```suggestion
  <div *ngIf="customIcon" [attr.class]="customIcon"></div>    
```

File: components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ export class SidebarEntryComponent {
   @Input() iconRotation: number;
   // (optional) if true, opens the link in a new page
   @Input() openInNewPage: boolean;
+  // (optional) shows custom icon
```
Comment (msorens) [inline components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.ts:16]: 👍 Good, but please add a similar line above `icon` on L9, something like:
```
// (optional) if present displays the standard icon
 @Input() icon: string;
```

Also, suggest putting the two declarations next to each other here (`icon` and `customIcon`) and please add a further comment that distinguishes the two. That is, when do I use `icon` and when do I use `customIcon` ? Like the casual reader, I actually do not know the answer as I write this. I would have to dig into the code to understand that--but let's help the reader so they do not have to. 😁 
(And a further question, coming from not yet digging into the code: do we really need both of them?)

File: components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.html (hunk 1 above)
Comment (sachin-msys) [reply components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.html]: @msorens nice catch :1st_place_medal: , incorporated :+1:

File: components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.ts (hunk 2 above)
Comment (sachin-msys) [reply components/automate-ui/src/app/components/sidebar-entry/sidebar-entry.component.ts]: @msorens yes we need both of them. `icon` is used to show material icon and `customIcon` is used when there is no respective material icon is available. I took reference from Waiver (#3089).

File: components/automate-ui/src/styles.scss [hunk 3]
```diff
@@ -342,6 +342,29 @@ input {
...
+
+.chef-servers-icon:before {
+  color: $chef-darkest;
+  content: 'C';
+  font-size: 9px;
+  font-weight: bolder;
+  position: absolute;
+  right: 4px;
+  text-align: center;
+  top: -3px;
```
Comment (SEAjamieD) [inline components/automate-ui/src/styles.scss:365]: [nit] @susanev / @jonong1972 / @apriofrost  - not sure who this falls under.

When I spin this up - it looks off center to me and `top` should be at `-4px` maybe?

Comment (sachin-msys) [reply components/automate-ui/src/styles.scss:365]: custom-c-4

Comment (susanev) [reply components/automate-ui/src/styles.scss:365]: yah, noticed this when the waivers one was added as well. ideally id love for them to be centered, but if that isn't easy thats okay too.

Comment (SEAjamieD) [reply components/automate-ui/src/styles.scss:365]: @sachin-msys - there must be some browser differences in play.  I get the opposite of what you see with those settings.  🤷‍♂️   

No need for this to be a blocker.  I'll toy with it in my spare time soon.

**Pull request merged**